src/backend/TrafficCourts/Staff.Service/Authentication/ConfigureJwtBearerOptions.cs

```diff
@@ -14,7 +18,49 @@ public ConfigureJwtBearerOptions(IConfiguration configuration)
 
     public void Configure(JwtBearerOptions options)
     {
-        options.Authority = _configuration["Jwt:Authority"];
-        options.Audience = _configuration["Jwt:Audience"];
+        // https://github.com/dotnet/aspnetcore/blob/v6.0.2/src/Security/Authentication/JwtBearer/samples/JwtBearerSample/Startup.cs
+        _configuration.Bind("Jwt", options);
+
+        options.Events = new JwtBearerEvents
+        {
+            OnTokenValidated = async context => await OnTokenValidatedAsync(context)
+        };
+    }
+
+    private static Task OnTokenValidatedAsync(TokenValidatedContext context)
+    {
+        if (context.SecurityToken is JwtSecurityToken accessToken
+                && context?.Principal?.Identity is ClaimsIdentity identity
+                && identity.IsAuthenticated)
+        {
+            identity.AddClaim(new Claim(ClaimTypes.Name, accessToken.Subject));
+            FlattenRealmAccessRoles(identity);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Flattens the Realm Access claim, as Microsoft Identity Model doesn't support nested claims
+    /// </summary>
+    private static void FlattenRealmAccessRoles(ClaimsIdentity identity)
```

Comment:
This code was taken from [Prime's AuthenticationSetup.cs](https://github.com/bcgov/moh-prime/blob/7d74974deda7f343d7d77c2225590491fb088869/prime-dotnet-webapi/Configuration/Auth/AuthenticationSetup.cs) class.